.github/workflows/perf.yml

```diff
@@ -0,0 +1,85 @@
+# 5. Click 'Create access key' to get the access key ID and secret access key
+# 6. Go to https://github.com/libp2p/test-plans/settings/secrets/actions
+# 7. Click 'New repository secret', set the name to 'PERF_AWS_SECRET_ACCESS_KEY', and paste the secret access key from step 5
+# 8. Click 'New repository secret', set the name to 'PERF_SSH_PRIVATE_KEY', and paste the private key from step 1
+# 9. Go to https://github.com/libp2p/test-plans/settings/variables/actions
+# 10. Click 'New repository variable', set the name to 'PERF_AWS_ACCESS_KEY_ID', and paste the access key ID from step 5
+
+on:
+  workflow_dispatch:
+
+jobs:
+  perf:
+    name: Perf
+    runs-on: ubuntu-latest
+    timeout-minutes: 40
+    defaults:
+      run:
+        shell: bash
+        working-directory: perf
+    env:
+      AWS_ACCESS_KEY_ID: ${{ vars.PERF_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.PERF_AWS_SECRET_ACCESS_KEY }}
+      TF_IN_AUTOMATION: 1
+      TF_INPUT: 0
+      TF_VAR_common_enabled: false
+      TF_VAR_region_enabled: false
+      TF_VAR_ephemeral_enabled: true
+    steps:
+      - name: Configure SSH
+        uses: webfactory/ssh-agent@d4b9b8ff72958532804b70bbe600ad43b36d5f2e # v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.PERF_SSH_PRIVATE_KEY }}
+      - name: Configure git
+        run: |
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com>"
+          git config --global user.name "${GITHUB_ACTOR}"
+      - name: Configure terraform
+        uses: hashicorp/setup-terraform@633666f66e0061ca3b725c73b2ec20cd13a8fdd1 # v2.0.3
+      - name: Checkout test-plans
+        uses: actions/checkout@v3
+        with:
+          repository: libp2p/test-plans
+          ref: ${{ github.ref }}
+      - name: Init terraform
+        run: terraform init
+        working-directory: perf/terraform
+      - name: Apply terraform
+        run: terraform apply -auto-approve
+        working-directory: perf/terraform
+      - id: server
+        name: Retrieve server's IP
+        run: echo "ip=$(terraform output -raw server_ip)"
+      - id: client
+        name: Retrieve client's IP
+        run: echo "ip=$(terraform output -raw client_ip)"
+      - name: Download dependencies
+        run: npm ci
+        working-directory: perf/runner
+      - name: Run tests
+        env:
+          SERVER_IP: ${{ steps.server.outputs.ip }}
+          CLIENT_IP: ${{ steps.client.outputs.ip }}
+        run: npm run start -- --client-public-ip $CLIENT_IP --server-public-ip $SERVER_IP
+        working-directory: perf/runner
+      - name: Archive results
+        uses: actions/upload-artifact@v2
+        with:
+          name: results
+          path: perf/runner/benchmark-results.json
+      - name: Push
+        run: |
+          git add perf/runner/benchmark-results.json
+          git commit -m "perf: update benchmark results"
+          git push
```

Comment:
> I don't think this is necessary. For the sake of simplicity, I suggest only pushing the results to the branch of the pull request. Gives us a single source-of-truth, fully version controlled.

Done! This will push benchmark-results.json to the branch from which the workflow was triggered. It will push it using the name of the person who triggered the workflow.